stellargraph/connector/neo4j/mapper.py

```diff
@@ -169,23 +131,25 @@ def sample_features(self, head_nodes, batch_num):
             where num_sampled_at_layer is the cumulative product of `num_samples`
             for that layer.
         """
-        nodes_per_hop = self.sampler.run(
-            self.neo4j_graphdb, nodes=head_nodes, n=1, n_size=self.num_samples
-        )
-        node_type = self.head_node_types[0]
+        nodes_per_hop = self.sampler.run(nodes=head_nodes, n=1, n_size=self.num_samples)
+
+        batch_nodes = np.concatenate(nodes_per_hop)
+        batch_features = self.graph.node_features(batch_nodes)
```

Comment:
> this is currently being done in the demo notebooks, I think it's currently quite a lot slower without the caching for such a small graph.

You mentioned this in sprint close, but I don't recall if you mentioned some numbers (and, it'd be nice to write them down for reference). Do you happen to remember approximately what the slowdown was?

Reply:
After commenting out the caching, I'm getting
```
Epoch 1/20
6/6 - 174s - loss: 1.8612 - acc: 0.2370 - val_loss: 1.7212 - val_acc: 0.3790
Epoch 2/20
6/6 - 172s - loss: 1.6545 - acc: 0.4926 - val_loss: 1.5768 - val_acc: 0.4413
```
compared to 6 seconds per epoch, so up to ~30 x slower with Cora.

Probably also quite heavily dependent on these parameters:
```
batch_size = 50
num_samples = [10, 5]
```